### PR TITLE
Use JsType+Global.JSON instead AutoBean and fix gwt-maven-plugin

### PR DESCRIPTION
**Pull request by ibaca**
Version rc-7 requires type:gwt-lib to correctly configure dependencies using devmode goal. And also, I have added a gwt maven plugin configuration to the root pom, I'm not sure how you are using devmode right now, but I think that you need to install before you can use it, isn't it? Now you can just use gwt:devmode in the root module and it just loads 3 examples at the same time.

I'm not sure if you really want to use AutoBean on purpose, but I think that if you are using elemental2 hehe you better do it all without "old gwt" magic. Using JsType+Global.JSON is the new AutoBean 😛.

### CODE REVIEW

File: gin/pom.xml [hunk 1]
```diff
@@ -41,6 +41,7 @@
         <dependency>
             <groupId>org.jboss.gwt.elemento</groupId>
             <artifactId>elemento-core</artifactId>
+            <type>gwt-lib</type>
```
Comment (ibaca) [inline gin/pom.xml]: this is not required, but as you are defining the version in the parent pom, this makes this version used by this dependency. You can instead remove the type and use <version>${project.version}</version>.

File: pom.xml [hunk 2]
```diff
@@ -102,6 +102,7 @@
                 <groupId>org.jboss.gwt.elemento</groupId>
                 <artifactId>elemento-core</artifactId>
                 <version>${project.version}</version>
+                <type>gwt-lib</type>
```
Comment (ibaca) [inline pom.xml]: this fixes devmode... see https://github.com/tbroyer/gwt-maven-plugin/issues/86

File: pom.xml [hunk 3]
```diff
@@ -255,6 +256,19 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>net.ltgt.gwt.maven</groupId>
+                <artifactId>gwt-maven-plugin</artifactId>
+                <inherited>false</inherited>
```
Comment (ibaca) [inline pom.xml]: not inherited! this is only used during development to run devmode.

**Comment by remarkNS** [comment]
+1 @ibaca. Replacing AutoBean with JsType+Global.JSON is great idea.

**Comment by ibaca** [comment]
Uhm, I'm going to split this PR in 2, 1.for autobean and another to the plugin, because the plugin configuration is no very IDE friendly, and because I just see that you use develop branch instead of master branch.

**Pull request closed without merging**